app.rb

```diff
@@ -0,0 +1,126 @@
+# file: app.rb
+
+require_relative './lib/item_repository'
+require_relative './lib/item'
+require_relative './lib/order_repository'
+require_relative './lib/database_connection'
+
+class Application
+
+  # The Application class initializer
+  # takes four arguments:
+  #  * The database name to call `DatabaseConnection.connect`
+  #  * the Kernel object as `io` (so we can mock the IO in our tests)
+  #  * the ItemRepository object (or a double of it)
+  #  * the OrderRepository object (or a double of it)
+  def initialize(database_name, io, item_repository, order_repository)
+    DatabaseConnection.connect(database_name)
+    @io = io
+    @item_repository = item_repository
+    @order_repository = order_repository
+  end
+
+  def run
```

Comment:
Love the clarity of the run method, referring to other created methods within the class